src/manager.js

```diff
@@ -482,18 +509,47 @@ AdManager.prototype.loadAds = function(element, updateCorrelator) {
   }
 
   for (var i = 0; i < ads.length; i++) {
-    var thisEl = ads[i];
-
-    if ((thisEl.getAttribute('data-ad-load-state') === 'loaded') || (thisEl.getAttribute('data-ad-load-state') === 'loading')) {
+    var thisEl = ads[i],
+      slot,
+      adUnitConfig,
+      activeSizes,
+      gptSlotSizes,
+      adUnitSizes;
+
+	if ((thisEl.getAttribute('data-ad-load-state') === 'loaded') || (thisEl.getAttribute('data-ad-load-state') === 'loading')) {
       continue;
     }
 
-    var slot = this.configureAd(thisEl);
+    slot = this.configureAd(thisEl);
 
     if (slot && slot.eagerLoad) {
       slotsToLoad.push(slot);
     }
 
+    if (this.options.amazonEnabled) {
```

Comment:
I'd like to separate all of the functionality like this specific to Amazon into a separate method outside of `#loadAds`, mainly because it makes it helps make it clearer the intent of this block

Reply:
@spra85 That makes sense, i will move it into a function.

Reply:
@spra85 Just added this.